crates/utils/re_arrow_combinators/src/semantic.rs

```diff
@@ -62,3 +65,63 @@ impl<O1: OffsetSizeTrait, O2: OffsetSizeTrait> Transform for BinaryToListUInt8<O
         Ok(list)
     }
 }
+
+/// Converts `StructArray` of timestamps with `seconds` (i64) and `nanos` (i32) fields
+/// to `Int64Array` containing the corresponding total nanoseconds timestamps.
+#[derive(Default)]
+pub struct TimeSpecToNanos {}
```

Comment:
I called this `TimeSpecToNanos` now, following the common Linux time type https://man7.org/linux/man-pages/man3/timespec.3type.html.

(I had `SecondsNanosToNanos` before but that looked awful)